File: ABC/ABC253/B.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int h,w; cin >> h >> w;
    string s[h];
    for(int i=0;i<h;i++) cin >> s[i];
    int ax=-1, ay, bx, by;
    for(int i=0;i<h;i++) {
        for(int j=0;j<w;j++) {
            if(s[i][j] == 'o') {
                if(ax == -1) {
                    ax = i;
                    ay = j;
                } else {
                    bx = i;
                    by = j;
                }
            }
        }
    }
    int ans = abs(ax - bx) + abs(ay - by);
    cout << ans << endl;
}

File: ABC/ABC253/b.rs
use proconio::input;
use std::cmp::*;
use proconio::marker::Chars;

fn main() {
    input!{
        h: usize,
        w: usize,
        s: [Chars; h],
    }

    let mut x1: i32 = -1;
    let mut x2: i32 = -1;
    let mut y1: i32 = -1;
    let mut y2: i32 = -1;

    for i in 0..h {
        for j in 0..w {
            if s[i][j] == 'o' {
                if x1 == -1 {
                    x1 = i as i32;
                    y1 = j as i32;
                } else {
                    x2 = i as i32;
                    y2 = j as i32;
                }
            }
        }
    }

    let ans = (x1 - x2).abs() + (y1 - y2).abs();
    println!("{}", ans);
}